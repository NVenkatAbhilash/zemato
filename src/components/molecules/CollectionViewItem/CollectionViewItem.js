import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Grid } from "@material-ui/core";
import Typography from "../../atoms/Typography/Typography";
//import foodImg from "./../../../resources/images/foodItem2.jpg";
//import foodImg from "./foodImg.png";
import ArrowRightIcon from "@material-ui/icons/ArrowRight";

const useStyles = makeStyles({
  root: {
    borderRadius: "15px",
    height: "480px",
    width: "100%",
    backgroundSize: "cover",
    backgroundPosition: "center",
    color: "white",
    cursor: "pointer",
  },
  small: {
    height: "250px",
  },
  content: {
    padding: "20px",
  },
  arrowIcon: {
    position: "relative",
    top: "8px",
  },
});

const CollectionViewItem = ({
  mode,
  foodImg,
  special,
  placesCount,
  ...props
}) => {
  const classes = useStyles();

  let rootClass = classes.root;

  if (mode === "small") {
    rootClass += " " + classes.small;
  }

  return (
    <Grid
      container
      alignItems="flex-end"
      className={rootClass}
      style={{
        backgroundImage: `linear-gradient(rgba(0,0,0, 0.1), rgb(0, 0, 0, .6)), url(${foodImg})`,
      }}
      data-testid="main"
      {...props}
    >
      <Grid item className={classes.content}>
        <Typography data-testid="special" variant="h6">
          {special}
        </Typography>
        <Typography variant="subtitle2" data-testid="place">
          {placesCount} Places
          <ArrowRightIcon className={classes.arrowIcon} fontSize="small" />
        </Typography>
      </Grid>
    </Grid>
  );
};

export default CollectionViewItem;
