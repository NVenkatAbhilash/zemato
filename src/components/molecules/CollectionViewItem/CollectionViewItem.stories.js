import React from "react";
import CollectionViewItem from "./CollectionViewItem";
import collection1 from "./../../../resources/images/collection2.jpg";


export default {
  title: "molecules/CollectionViewItem",
  component: CollectionViewItem,
};

const Template = ({ ...args }) => <CollectionViewItem {...args} />;

export const Medium = Template.bind({});
Medium.args = {
  special: "What's Special",
  placesCount: 12,
  foodImg : collection1,
};

export const Small = Template.bind({});
Small.args = {
  special: "What's Special",
  placesCount: 12,
  foodImg : collection1,
  mode: "small",
};
