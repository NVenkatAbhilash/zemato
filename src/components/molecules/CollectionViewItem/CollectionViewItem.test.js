import React from "react";
import ReactDOM from "react-dom";
import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import CollectionViewItem from "./CollectionViewItem";
import collection1 from "./../../../resources/images/collection2.jpg";
import renderer from 'react-test-renderer';

it('renders CollectionViewItem snapshot correctly', () => {
  const args = {
    special: "What's Special",
    placesCount: 12,
    foodImg: collection1,
  };

  const tree = renderer
    .create(<CollectionViewItem {...args}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders CollectionViewItem snapshot correctly 2', () => {
  const tree = renderer
    .create(<CollectionViewItem/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders CollectionViewItem without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<CollectionViewItem />, div);
});

it("renders CollectionViewItem correctly", () => {
  const args = {
    special: "What's Special",
    placesCount: 12,
    foodImg: collection1,
  };
  const { getByTestId } = render(<CollectionViewItem {...args}/>);
  expect(getByTestId("special")).toHaveTextContent(args.special);
  expect(getByTestId("place")).toHaveTextContent(args.placesCount+ " Places");
});

it("renders CollectionViewItem correctly when mode='small'", () => {
  const args = {
    special: "What's Special",
    placesCount: 12,
    foodImg: collection1,
  };
  const { getByTestId } = render(<CollectionViewItem mode="small" {...args}/>);
  expect(getByTestId("special")).toHaveTextContent(args.special);
  expect(getByTestId("place")).toHaveTextContent(args.placesCount+ " Places");
});
