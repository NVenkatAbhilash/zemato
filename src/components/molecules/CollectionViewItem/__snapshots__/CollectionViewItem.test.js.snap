// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CollectionViewItem snapshot correctly 1`] = `
<div
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-align-items-xs-flex-end"
  data-testid="main"
  style={
    Object {
      "backgroundImage": "linear-gradient(rgba(0,0,0, 0.1), rgb(0, 0, 0, .6)), url(collection2.jpg)",
    }
  }
>
  <div
    className="MuiGrid-root makeStyles-content-3 MuiGrid-item"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6"
      data-testid="special"
    >
      What's Special
    </h6>
    <h6
      className="MuiTypography-root MuiTypography-subtitle2"
      data-testid="place"
    >
      12
       Places
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root makeStyles-arrowIcon-4 MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 17l5-5-5-5v10z"
        />
      </svg>
    </h6>
  </div>
</div>
`;

exports[`renders CollectionViewItem snapshot correctly 2 1`] = `
<div
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-align-items-xs-flex-end"
  data-testid="main"
  style={
    Object {
      "backgroundImage": "linear-gradient(rgba(0,0,0, 0.1), rgb(0, 0, 0, .6)), url(undefined)",
    }
  }
>
  <div
    className="MuiGrid-root makeStyles-content-3 MuiGrid-item"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6"
      data-testid="special"
    />
    <h6
      className="MuiTypography-root MuiTypography-subtitle2"
      data-testid="place"
    >
       Places
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root makeStyles-arrowIcon-4 MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 17l5-5-5-5v10z"
        />
      </svg>
    </h6>
  </div>
</div>
`;
